builtin/logical/aws/secret_access_keys.go

```diff
@@ -123,6 +173,16 @@ func (b *backend) secretAccessKeysRenew(
 func secretAccessKeysRevoke(
 	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
 	// Get the username from the internal data
+	isSTS, ok := req.Secret.InternalData["is_sts"]
+	if !ok {
+		return nil, fmt.Errorf("secret is missing is_sts internal data")
+	}
+
+	// STS cleans up after itself
+	if isSTS.(bool) {
```

Comment:
This can panic if, for some reason, it's not a bool. I think the best thing to do would be instantiate a bool var (with it defaulting to nil), then use the `isSTS, ok := req.Secret.InternalData["is_sts"]` call to drop into this check if `ok` is true. Use the dual-value version here. So something like (not tested/checked):
```
stsInt, ok := req.Secret.InternalData["is_sts"]
if ok {
  isSTS, ok = stsInt.(bool)
  if ok {
    if isSTS {
      return nil, nil
    } // else, it was set, but not to true, so treat it as IAM by falling through
  } else {
    return nil, fmt.Errorf("secret has is_sts but value could not be understood")
  }
}